synapse/cores/common.py

```diff
@@ -2600,8 +2600,9 @@ def _tufosByInetCidr(self, prop, valu, limit=None):
 
         ipv4str, cidr = valu.split('/', 1)
         ipv4addr, _ = s_datamodel.getTypeParse('inet:ipv4', ipv4str)
-        mask = (2 ** (32 - int(cidr)))
-        ipv4addr &= ~mask
+        shift = 32 - int(cidr)
+        ipv4addr = ipv4addr >> shift << shift
+        mask = (2 ** shift)
```

Comment:
Good call on the perf stuff though.  Actually moving the cidr upper/lower bound gives us a utility function for computing the upper/lower values; which can be turned around and registered as a type cast.  While not the FASTEST approach, its pretty versatile and prevents the dynamic generation.

Perf results of a few different implementations cab be seen here https://gist.github.com/vEpiphyte/8b8c2a83ffba4885adc0f05891b16481